lib/_stream_wrap.js

```diff
@@ -111,10 +138,81 @@ StreamWrap.prototype.write = function write(req, bufs) {
           errCode = uv.UV_EPIPE;
       }
 
-      self._handle.doAfterWrite(req);
-      self._handle.finishWrite(req, errCode);
+      handle.doAfterWrite(req);
+      handle.finishWrite(req, errCode);
     });
   }
 
   return 0;
 };
+
+function QueueItem(type, req) {
+  this.type = type;
+  this.req = req;
+  this.prev = this;
+  this.next = this;
+}
+
+StreamWrap.prototype._enqueue = function enqueue(type, req) {
+  const item = new QueueItem(type, req);
+  if (this._queue === null) {
+    this._queue = item;
```

Comment:
Can this be changed to `_list` or similar? At a quick glance the naming would tell me that it's a queue, event though this is a double-linked list. Which makes the code above `_dequeue(item)` confusing, since usually that would denote we're grabbing and item from the queue. Instead of removing `item` from the linked list.